### PR TITLE
Write out explicit impropers

### PR DESCRIPTION
**Pull request by ChayaSt**
This PR includes the changes to write out impropers explicitly for OpnenMM `ffxml` files.

### CODE REVIEW

File: parmed/charmm/parameters.py [hunk 1]
```diff
@@ -811,6 +811,11 @@ def read_topology_file(self, tfile):
                             atom = Atom(name=name, type=type, charge=charge)
                             group.append(atom)
                             res.add_atom(atom)
+                        elif line[:6].upper() == 'DELETE':
+                            words = line.split()
+                            name = words[2].upper()
+                            atom = Atom(name=name)
+                            res.delete.append(atom)
```
Comment (swails) [inline parmed/charmm/parameters.py]: Just add the `name`, not an `Atom` instance.

**Comment by ChayaSt** [comment]
This PR addresses https://github.com/choderalab/openmm-forcefields/issues/33.

To address the ambiguity that arises when more than one wild card is present in an improper type, this PR writes out the impropers explicitly.

For every `IMPR` entry in `RESI`, it finds the improper type and writes it out explicitly with the corresponding atom types.

**Comment by swails** [comment]
run jenkins

**Comment by ChayaSt** [comment]
@swails, should I also add this change [here](https://github.com/ParmEd/ParmEd/blob/master/parmed/charmm/psf.py#L630-L646)?

That scheme leads to the wrong matched improper. see this comment:
https://github.com/ParmEd/ParmEd/issues/866#issue-224269731

**Comment by swails** [comment]
Sounds good!

**Comment by ChayaSt** [comment]
This also fixes https://github.com/ParmEd/ParmEd/issues/866

**Comment by swails** [comment]
run jenkins

**Comment by swails** [comment]
This PR is out-of-date with the master branch.  I submitted a PR that merges master and resolves conflicts.  Once you merge that (and this gets tested), then this should be good to merge.

**Comment by swails** [comment]
run jenkins

**Comment by swails** [comment]
run jenkins

**Comment by swails** [comment]
@ChayaSt -- is this good to merge?

**Comment by ChayaSt** [comment]
@swails, we can merge this but it will need more work once the patches are complete. I will open another PR then.

**Pull request merged**